lib/apiservers/engine/backends/system.go

```diff
@@ -15,31 +15,148 @@
 	"github.com/docker/engine-api/types/filters"
+	"github.com/docker/go-connections/sockets"
 )
 
 type System struct {
 	ProductName string
 }
 
+type ContainerStatus struct {
+	count      int
+	numRunning int
+	numStopped int
+	numPaused  int
+}
+
+var (
+	etcReleaseFile    = "/etc/os-release"
+	usrlibReleaseFile = "/usr/lib/os-release"
+)
+
 func (s *System) SystemInfo() (*types.Info, error) {
-	Driver := "Portlayer Storage"
-	IndexServerAddress := "https://index.docker.io/v1/"
-	ServerVersion := "0.0.1"
-	Name := "VIC"
+	// Set VIC specific data
+	Driver := s.ProductName + " Backend Engine"
+	IndexServerAddress := "registry-1.docker.io"
+	ServerVersion := "0.3"
+
+	// Use docker pkgs to get some system data
+	kernelVersion := "<unknown>"
+	if kv, err := kernel.GetKernelVersion(); err != nil {
+		log.Warnf("Could not get kernel version: %v", err)
+	} else {
+		kernelVersion = kv.String()
+	}
+
+	operatingSystem := getOperatingSystem()
+
+	meminfo, err := system.ReadMemInfo()
+	if err != nil {
+		log.Errorf("Could not read system memory info: %v", err)
+	}
 
+	// Check if portlayer server is up
+	plClient := PortLayerClient()
+
+	systemStatus := make([][2]string, 1)
+	systemStatus[0][0] = s.ProductName + " Backend Engine"
+	if pingPortlayer(plClient) {
+		systemStatus[0][1] = "RUNNING"
+	} else {
+		systemStatus[0][1] = "STOPPED"
+	}
+
+	// Retrieve info about the VCH from Exec port layer
+	//FIXME: Write the portlayer code to do this
+
+	// Retrieve number of images from storage port layer
+	numImages, err := getImageCount(plClient)
+	if err != nil {
+		log.Info("System.SytemInfo unable to get image count.")
+	}
+
+	// Retieve container status from port layer
+	containerStatus, err := getContainerStatus(plClient)
+	if err != nil {
+		log.Info("System.SytemInfo unable to get global status on containers")
+	}
+
+	// Build up the struct that the Remote API and CLI wants
 	info := &types.Info{
 		Driver:             Driver,
 		IndexServerAddress: IndexServerAddress,
 		ServerVersion:      ServerVersion,
-		Name:               Name,
+		ID:                 s.ProductName,
+		Containers:         containerStatus.count,
+		ContainersRunning:  containerStatus.numRunning,
+		ContainersPaused:   containerStatus.numPaused,
+		ContainersStopped:  containerStatus.numStopped,
+		Images:             numImages,
+		SystemStatus:       systemStatus,
+		Debug:              (os.Getenv("DEBUG") != ""),
+		NGoroutines:        runtime.NumGoroutine(),
+		SystemTime:         time.Now().Format(time.RFC3339Nano),
+		ExecutionDriver:    Driver,
+		LoggingDriver:      "json-file",
+		CgroupDriver:       "N/A",
+		// FIXME: Get this info once we have event listening service
+		//	NEventsListener    int
+		DockerRootDir:    "N/A",
+		ClusterStore:     "",
+		ClusterAdvertise: "",
+		HTTPProxy:        sockets.GetProxyEnv("http_proxy"),
+		HTTPSProxy:       sockets.GetProxyEnv("https_proxy"),
+		NoProxy:          sockets.GetProxyEnv("no_proxy"),
```

Comment:
Why do this via the environment at all - it's not likely to be set in the appliance.
If we need a proxy then it should be in the VCH config struct so that imagec can make use of it as well.